src/service/routing/spec.clj

```diff
@@ -31,41 +43,14 @@
 (s/def ::routes (s/coll-of ::route-object :kind sequential?))
 (s/def ::direction (s/keys :req-un [::code] :opt-un [::waypoints ::routes]))
 
-(def coordinate-regex #"(-?\d+(\.\d+)?),(-?\d+(\.\d+)?)(;(-?\d+(\.\d+)?),(-?\d+(\.\d+)?))+")
-(s/def ::coordinate-regex (s/and string? #(re-matches coordinate-regex %)))
-
-(defn ->coordinates
-  [text]
-  (let [pairs (str/split text #";")]
-    (for [coords pairs]
-      (map edn/read-string (str/split coords #",")))))
-
-;(->coordinates "-122.42,37.78;-77.03,38.91")
-
-(def routes
-  (context "/spec" []
-    :tags ["spec"]
-    :coercion :spec
-
-    (GET "/direction/:coordinates" []
-      :summary "direction with clojure.spec"
-      :path-params [coordinates :- ::coordinate-regex]
-      :return ::direction
-      (ok (let [coords (map zipmap (repeat [:lon :lat])
-                                   (->coordinates coordinates))]
-            (dir/direction (gen/generate (g/graph 1000))
-              :coordinates coords))))))
-
-    ; (context "/data-plus" []
-    ;   (resource
-    ;     {:post
-    ;      {:summary "data-driven plus with clojure.spec"
-    ;       :parameters {:body-params (s/keys :req-un [::x ::y])}
-    ;       :responses {200 {:schema ::total-map}}
-    ;       :handler (fn [{{:keys [x y]} :body-params}]
-    ;                  (ok {:total (+ x y)}))}}))
-
-
-;;;; TEST @mehdi
-;(expound/expound-str ::direction (dir/direction (gen/generate (g/graph 1000)) :coordinates [{:lon 1 :lat 2} {:lon 3 :lat 4}]))
-;(s/explain ::direction (dir/direction (gen/generate (g/graph 1000)) :coordinates [{:lon 1 :lat 2} {:lon 3 :lat 4}]))
+(s/def ::coordinates-raw
+  (s/and string? #(re-matches coordinate-regex %)
+         (s/conformer parse-coordinates)
```

Comment:
@mehdisadeghi I am actually not sure if this will work. I know that spec supports this but I dont know if compojure-api is already that advance to support it.

Please take it with a bit of salt ;)